### DISCUSSION TITLE
Starting HAP Server issues

### DISCUSSION THREAD

**Discussion by kned9x** [HomeSpan/HomeSpan#666] [Q&A] [answered]
Dear community,

I'm trying to run HomeSpan on an esp32-s2. However I cannot pair the Bridge I've set up with my iPhone.
So I checked with example 1 and it seems that it gets stuck while 'starting HAP Server on port 80 ...'.
Normally it should say 'Device not paired ...', souldn't it ?
Below attached my serial monitor, if I enter any input, for example W or H nothing happens.

Thanks for any help. 

`
16:05:17.381 -> ************************************************************
16:05:17.381 -> Welcome to HomeSpan!
16:05:17.381 -> Apple HomeKit for the Espressif ESP-32 WROOM and Arduino IDE
16:05:17.413 -> ************************************************************
16:05:17.413 -> 
16:05:17.413 -> ** Please ensure serial monitor is set to transmit <newlines>
16:05:17.413 -> 
16:05:17.413 -> Message Logs:     Level 0
16:05:17.413 -> Status LED:       Pin -  *** WARNING: Status LED Pin is UNDEFINED
16:05:17.413 -> Device Control:   Pin -  *** WARNING: Device Control Pin is UNDEFINED
16:05:17.413 -> Sketch Version:   n/a
16:05:17.413 -> HomeSpan Version: 1.8.0
16:05:17.413 -> Arduino-ESP Ver.: 2.0.11
16:05:17.413 -> ESP-IDF Version:  4.4.5
16:05:17.447 -> ESP32 Chip:       ESP32-S2 Rev 0 single-core 4MB Flash
16:05:17.447 -> ESP32 Board:      esp32s2
16:05:17.447 -> PWM Resources:    8 channels, 4 timers, max 14-bit duty resolution
16:05:17.447 -> Sodium Version:   1.0.12-idf  Lib 9.4
16:05:17.447 -> MbedTLS Version:  mbed TLS 2.28.3
16:05:17.447 -> Sketch Compiled:  Oct 21 2023 16:04:42
16:05:17.447 -> Partition:        app0
16:05:17.447 -> MAC Address:      84:F7:03:DB:41:1A
16:05:17.481 -> 
16:05:17.481 -> Device Name:      HomeSpan LightBulb
16:05:17.481 -> 
16:05:17.481 -> 
16:05:17.481 -> *** HomeSpan Info ***
16:05:17.481 -> 
16:05:17.481 -> ➤ Accessory:  AID=1
16:05:17.481 ->    ➟ Service AccessoryInformation:  IID=1, UUID="3E"
16:05:17.481 ->       ⇨ Characteristic Identify(0):  IID=2, UUID="14", Perms=PW
16:05:17.481 ->    ➟ Service LightBulb:  IID=3, UUID="43"
16:05:17.481 ->       ⇨ Characteristic On(0):  IID=4, UUID="25", Perms=PR+PW+EV
16:05:17.481 -> 
16:05:17.481 -> Configured as Bridge: NO
16:05:17.481 -> 
16:05:17.481 -> Database Validation:  Warnings=0, Errors=0
16:05:17.514 -> 
16:05:17.514 -> Service                             UUID         AID  IID  Update  Loop  Button  Linked Services
16:05:17.514 -> ------------------------------  --------  ----------  ---  ------  ----  ------  ---------------
16:05:17.514 -> AccessoryInformation                  3E           1    1      NO    NO      NO  -
16:05:17.514 -> LightBulb                             43           1    3      NO    NO      NO  -
16:05:17.514 -> 
16:05:17.514 -> *** End Info ***
16:05:17.551 -> 
16:05:17.551 -> Accessory ID:      59:F4:7A:8A:3C:AE                               LTPK: 087268D7C2E952423AB9BFDE840E966259CC27491C4053D79E46DC2DF524FDAA
16:05:17.551 -> Paired Controller: 5C85D290-A407-4DAE-AE9F-B5A7B6B69002   (admin)  LTPK: 6B6CFD90F511F1A3ABA5F15D232CC89D4689DD33AC7D2EA00AF43B6D4A0ACB0C
16:05:17.551 -> 
16:05:17.551 -> Accessory configuration number: 10
16:05:17.551 -> 
16:05:17.551 -> HomeSpan LightBulb is READY!
16:05:17.551 -> 
16:05:17.551 -> Trying to connect to CDMX2.  Waiting 1 sec...
16:05:18.648 -> WiFi Connected!  IP Address = 192.168.1.66
16:05:18.648 -> 
16:05:18.648 -> Starting MDNS...
16:05:18.648 -> 
16:05:18.648 -> HostName:      HomeSpan-59F47A8A3CAE.local:80
16:05:18.648 -> Display Name:  HomeSpan LightBulb
16:05:18.648 -> Model Name:    HomeSpan-ESP32
16:05:18.648 -> Setup ID:      HSPN
16:05:18.648 -> 
16:05:18.648 -> Starting HAP Server on port 80 supporting 14 simultaneous HomeKit Controller Connections...

`

**Comment by HomeSpan** [comment]
I've just checked by using Example 1 on an ESP32-S2 under both version 2.0.13 and the latest 2.0.14 Arduino-ESP32 Board Manager and everything works fine (pairing with iPhone as well as turning on/off the light).  Are you testing with a completely unmodified version of Example 1?  Also, please double-check that you have scrolling enabled on the Serial Monitor - I only ask because where you indicate things froze above is approximately a full screen of text.

**Reply by kned9x** [reply]
Hello, thank you for your swift response!
I ran a completely unmodified example 1 and wasn't able to connect.
Are you referring to 'toggle autoscroll' ? In that case it is turned on and I can also manually not scroll further down.
I also tried a second iPhone to pair it without success.
The board is normally powered by a 5V source but I also tried to power it only by USB, without success.
What confuses me most, is that it is able to connect to the WiFi and then gets stuck, so WiFi is not the issue apparently.
Kind regards.

**Comment by HomeSpan** [comment, marked as answer]
What happens if you run without pairing or even starting WiFi?  Does it respond keyboard input (for example, type 'd' or 's' into the Serial Monitor).

**Reply by kned9x** [reply]
It reacted and I was able to erase Wifi credentials and perform a factory reset. After setting it up again, now all working fine. Thank you!
